pkg/flow/internal/controller/loader.go

```diff
@@ -828,10 +832,9 @@ func multierrToDiags(errors error) diag.Diagnostics {
 	return diags
 }
 
-// If the definition of a module ever changes, update this.
-func (l *Loader) isModule() bool {
-	// Either 1 of these checks is technically sufficient but let's be extra careful.
-	return l.globals.OnExportsChange != nil && l.globals.ControllerID != ""
+// isRootController returns true if the loader is for the root flow controller.
```

Comment:
We considered separating the module vs non root controller concept but decided it wasn't worth the potential complexity of additional code paths. With the new `isRootController` note that the logical operator was inverted and usage was updated everywhere.